### ISSUE TITLE
`SoftBody3D` spatial attachments don't work in exported builds

### ISSUE THREAD

**Issue by papanouel** [godot-jolt/godot-jolt#930]
Hello!

First great thanks for all the work. I love godot, but without godot-jolt my game would not go far...

While implementing Softbodies for a basketball net, I noticed that there was a different between the desktop build and a mobile build (iOS and Android).
I have a basketball net (softbody) attached to a basketball rim (Node3D) through 32 pinned points. Each points have their spatial parent set to the meshinstance. My test consist on moving the MeshnInstance and see if the net follows or not. 

Runnning from the editor on my mac, the net does stay attached to the rim. However on mobile it does not.
FYI, I also tried attach the net to a rigid body, but same result.

2 possibilities:
1- I am doing something wrong, and the correct behaviour is what occuring on mobile
2- There's just a bug on mobile

Let me know if you see where I may have made a mistake

Mac/editor video
https://github.com/user-attachments/assets/bb3d8eef-2c49-4792-9fef-be810352d645

Phone video
https://github.com/user-attachments/assets/556c7d9c-86d8-43a4-8847-5e1da76dc2d9

Softbody setup
![image](https://github.com/user-attachments/assets/32b69677-b981-43a9-a965-84ba18088e21)

**Comment by mihe** [comment]
Thank you for reporting this. Just based on your description and videos this sounds a lot like a bug of some sort. In general you should never experience wildly different behaviors across different platforms like that.

I'll see if I can reproduce it myself.

**Comment by mihe** [comment]
I can confirm that I am able to reproduce this on iOS. There seems to be something funky with the `SoftBody3D` node on the Godot side of things when it comes to the spatial attachments, but I'm not entirely sure what just yet.

In the meantime it appears that `PhysicsServer3D.soft_body_move_point` works just fine, which is what the spatial attachments in `SoftBody3D` use behind the scenes, so if you need a workaround in the meantime then that could be an option. Basically pin the points as normal but leave the attachments blank and move them yourself using that method.

**Comment by mihe** [comment]
It turns out this applies to all exported projects, not only mobile ones, and happens with Godot Physics too.

**Comment by mihe** [comment]
It turns out this is a fairly long-standing issue in Godot, since before Godot 4.0-stable was released: godotengine/godot#82434.

Even if I manage to whip up a pull request for this I don't think it'll be merged for Godot 4.3-stable, given that it's meant to be released next week, so I'll put this on the back burner for a bit.

**Comment by mihe** [comment]
There seems to be a pending PR that fixes this issue already: godotengine/godot#86310

**Comment by papanouel** [comment]
Hello, 
Sorry for my late reply, I missed github notifications. Great thanks for the followup! Everything you said is 100% clear.
Just saw the pending PR. Things are looking great. Meanwhile, I'll follow your recommendation by using soft_body_move_point so that I don't have to modify my current Godot setup.

Thanks again for everything!

**Comment by mihe** [comment]
I'll leave this open until it gets fixed on the Godot side of things, in case more people run into it.

**Comment by papanouel** [comment]
For people Interested in a quick and "clean" workaround I ended up setting up the attachment in a script attached to my net (parent of my softbody) like this user did: https://github.com/godotengine/godot/issues/82434#issuecomment-1823876116

`# Called when the node enters the scene tree for the first time.
func _ready() -> void:
	softbody.set("attachments/0/point_index", 78)
	softbody.set("attachments/0/spatial_attachment_path", NodePath(".."))
	softbody.set("attachments/0/offset", Vector3(-0.262972, 0, 0.26296))
	softbody.set("attachments/1/point_index", 82)
	softbody.set("attachments/1/spatial_attachment_path", NodePath(".."))
	softbody.set("attachments/1/offset", Vector3(-0.309221, 0, 0.20661))`

So basically, set the attachment manually after the scene is loaded and ready.
You can copy the infos from the tscn file that contain the softbody.

**Comment by mihe** [comment]
The fix for this on the Godot side of things (godotengine/godot#86310) has now been merged, so I'll go ahead and close this.
